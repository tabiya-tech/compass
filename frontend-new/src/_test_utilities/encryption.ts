import { webcrypto as crypto } from "crypto";
import { EncryptionConfig } from "src/sensitiveData/config/encryptionConfig";

// Constraining the key size to 2048 or 4096 bits for RSA as the backend expect the 4096 in calculating the maximum size of the encrypted key.
export async function generateRSACryptoPairKey(keySize: 2048 | 4096) {
  return await crypto.subtle.generateKey(
    {
      name: EncryptionConfig.RSA.ALGORITHM,
      modulusLength: keySize,
      // 3 is a common exponent in practice
      publicExponent: new Uint8Array([1, 0, 1]),
      hash: EncryptionConfig.RSA.KEY_HASH_FN,
    },
    true, // Whether the key is extractable (i.e. can be used in exportKey)
    ["encrypt", "decrypt"] // the key can be used for both encryption and decryption
  );
}

function arrayBufferToBase64(buffer: ArrayBuffer): string {
  // @ts-ignore
  const binary = String.fromCharCode(...new Uint8Array(buffer));

  return Buffer.from(binary, "binary").toString("base64");
}

/**
 * Export a CryptoKey to a PEM formatted string (public key)
 *
 * @param key The CryptoKey to export as a PEM formatted string
 * @returns The PEM formatted string.
 */
export async function exportCryptoPublicKey(key: CryptoKey) {
  // Export the key to ArrayBuffer
  const exported = await crypto.subtle.exportKey(EncryptionConfig.RSA.FORMAT, key);

  const base64String = arrayBufferToBase64(exported);

  // add the PEM header and footer to make it a PEM formatted string.
  return `-----BEGIN PUBLIC KEY-----\n${base64String}\n-----END PUBLIC KEY-----`;
}

/**
 * Decrypt the encrypted data with the private key
 * using RSA Algorithm.
 *
 * @param privateKey
 * @param encryptedData
 */
export async function decryptWithRSA(privateKey: CryptoKey, encryptedData: ArrayBuffer) {
  return await crypto.subtle.decrypt(
    {
      name: EncryptionConfig.RSA.ALGORITHM,
    },
    privateKey,
    encryptedData
  );
}

export async function decryptWithAES(encryptedData: ArrayBuffer, iv: ArrayBuffer, key: ArrayBuffer) {
  try {
    const importedKey = await crypto.subtle.importKey(
      EncryptionConfig.AES.FORMAT,
      key,
      EncryptionConfig.AES.ALGORITHM,
      true,
      ["decrypt"]
    );

    return await crypto.subtle.decrypt(
      {
        name: EncryptionConfig.AES.ALGORITHM,
        iv,
        length: EncryptionConfig.AES.KEY_LEN,
        tagLength: EncryptionConfig.AES.TAG_LEN,
      },
      importedKey,
      encryptedData
    );
  } catch (e) {
    console.error(e);
    throw e;
  }
}
