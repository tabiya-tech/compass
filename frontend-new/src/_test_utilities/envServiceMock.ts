jest.mock("src/envService", () => ({
  getFirebaseAPIKey: jest.fn(() => "mock-api-key"),
  getFirebaseDomain: jest.fn(() => "mock-auth-domain"),
  getBackendUrl: jest.fn(() => "mock-backend-url"),
  getApplicationLoginCode: jest.fn(() => ""),
  getApplicationRegistrationCode: jest.fn(() => ""),
  getLoginCodeDisabled: jest.fn(() => "false"),
  getRegistrationDisabled: jest.fn(() => "false"),
  getMetricsEnabled: jest.fn(() => "true"),
  getMetricsConfig: jest.fn(() => ""),
  getCvUploadEnabled: jest.fn(() => "true"),
  getSocialAuthDisabled: jest.fn(() => "false"),
}));
