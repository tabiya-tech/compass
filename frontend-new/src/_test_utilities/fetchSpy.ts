import { ExtendedRequestInit } from "src/utils/customFetch/customFetch";

export function setupFetchSpy(
  expectedStatus: number,
  expectedResponseBody: string | object | undefined,
  contentType: "" | "application/json;charset=UTF-8"
): jest.SpyInstance {
  const responseBody =
    typeof expectedResponseBody === "string" ? expectedResponseBody : JSON.stringify(expectedResponseBody);
  const expectedResponse = new Response(responseBody, {
    status: expectedStatus,
    headers: { "Content-Type": contentType },
  });
  return jest.spyOn(window, "fetch").mockResolvedValue(expectedResponse);
}

// spy on authenticated fetch requests
export function setupAPIServiceSpy(
  expectedStatus: number,
  expectedResponseBody: string | object | undefined,
  contentType: "" | "application/json;charset=UTF-8"
): jest.SpyInstance {
  const responseBody =
    typeof expectedResponseBody === "string" ? expectedResponseBody : JSON.stringify(expectedResponseBody);
  const expectedResponse = new Response(responseBody, {
    status: expectedStatus,
    headers: { "Content-Type": contentType },
  });

  jest.spyOn(window, "fetch").mockResolvedValue(expectedResponse);
  return jest.spyOn(require("src/utils/customFetch/customFetch"), "customFetch");
}

export function expectCorrectFetchRequest(fetchSpy: jest.SpyInstance, expectedUrl: string, expectedConfig: ExtendedRequestInit) {
  expect(fetchSpy).toHaveBeenCalledTimes(1);
  const [actualUrl, actualConfig] = fetchSpy.mock.calls[0];
  expect(actualUrl).toBe(expectedUrl);
  expect(actualConfig.method).toBe(expectedConfig.method);

  // Compare headers if they exist
  if(actualConfig.headers) {
    // Handle headers comparison - convert Headers object to plain object if needed
    const actualHeaders = actualConfig.headers instanceof Headers
      ? Object.fromEntries(actualConfig.headers.entries())
      : actualConfig.headers;
    expect(actualHeaders).toEqual(
      expect.objectContaining(expectedConfig.headers),
    );
  }

  // Compare bodies by parsing JSON if they exist
  if (actualConfig.body && expectedConfig.body) {
    let actualBody: unknown;
    let expectedBody: unknown;
    
    try {
      actualBody = JSON.parse(JSON.stringify(actualConfig.body));
      expectedBody = JSON.parse(JSON.stringify(expectedConfig.body));
    } catch (e) {
      console.error('Error parsing request bodies:', e);
      throw new Error('Failed to parse request bodies for comparison');
    }
    expect(actualBody).toEqual(expectedBody);
  }

  // Check any extra properties if provided
  type ExtendedConfigKey = keyof ExtendedRequestInit;
  const extraProperties: ExtendedConfigKey[] = [
    'expectedStatusCode',
    'serviceName',
    'serviceFunction',
    'failureMessage',
    'expectedContentType'
  ] as ExtendedConfigKey[];

  extraProperties.forEach((key) => {
    if (expectedConfig[key] !== undefined) {
      expect(actualConfig[key]).toEqual(expectedConfig[key]);
    }
  });
}

