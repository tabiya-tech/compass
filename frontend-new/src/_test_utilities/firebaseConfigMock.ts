import { AuthProviderIds } from "src/auth/auth.types";

const auth = jest.fn(() => ({
  signInWithCustomToken: jest.fn(),
  signInWithEmailAndPassword: jest.fn(),
  createUserWithEmailAndPassword: jest.fn(),
  signOut: jest.fn(),
  GoogleAuthProvider: { PROVIDER_ID: AuthProviderIds.GOOGLE },
}));

export function mockFirebaseConfig() {
  jest.spyOn(require("src/auth/firebaseConfig"), "auth").mockImplementation(auth);
}
