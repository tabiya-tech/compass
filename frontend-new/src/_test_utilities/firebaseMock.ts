import { AuthProviderIds } from "src/auth/auth.types";

const auth = {
  currentUser: {
    getIdToken: jest.fn(),
    getTokenResult: jest.fn(),
  },
  onAuthStateChanged: jest.fn().mockReturnValue(jest.fn()),
  signInWithPopup: jest.fn(),
  GoogleAuthProvider: { PROVIDER_ID: AuthProviderIds.GOOGLE },
};

const firebase = {
  initializeApp: jest.fn(),
  auth: () => auth,
};

export default firebase;
