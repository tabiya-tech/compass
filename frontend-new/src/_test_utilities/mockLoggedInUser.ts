import { AuthContextValue, TabiyaUser, authContextDefaultValue } from "src/auth/AuthProvider";

export function mockLoggedInUser(mockedValues: Partial<AuthContextValue>) {
  let user: TabiyaUser = mockedValues.user || TestUser;
  jest.spyOn(require("src/auth/hooks/useAuthUser"), "useAuthUser").mockReturnValue({
    ...authContextDefaultValue,
    user: user,
    updateUser: (user: TabiyaUser) => null,
    updateUserByAccessToken: () => {},
    ...mockedValues,
  });
}

export const TestUser = {
  id: "0001",
  name: "Test User",
  email: "test@email.com",
};
