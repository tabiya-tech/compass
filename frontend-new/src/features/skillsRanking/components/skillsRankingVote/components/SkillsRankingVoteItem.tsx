import React from "react";
import { Theme, useTheme } from "@mui/material/styles";
import { Divider, useMediaQuery, Box, styled, Typography } from "@mui/material";
import PrimaryIconButton from "src/theme/PrimaryIconButton/PrimaryIconButton";
import { getBackgroundColor, getBorderStyle, getDividerColor } from "src/features/skillsRanking/utils/utils";

const uniqueId = "5271610c-0000-4000-8000-000000000000";

export const DATA_TEST_ID = {
  SKILLS_RANKING_VOTE_ITEM: `skills-ranking-vote-item-${uniqueId}`,
  SKILLS_RANKING_VOTE_ITEM_RADIO: `skills-ranking-vote-item-radio-${uniqueId}`,
  SKILLS_RANKING_VOTE_ITEM_DIVIDER: `skills-ranking-vote-item-divider-${uniqueId}`,
  SKILLS_RANKING_VOTE_ITEM_TOP_LABEL: `skills-ranking-vote-item-top-label-${uniqueId}`,
  SKILLS_RANKING_VOTE_ITEM_BOTTOM_LABEL: `skills-ranking-vote-item-bottom-label-${uniqueId}`,
};

interface SkillsRankingVoteItemProps {
  option: { value: string; percent: number };
  index: number;
  selectedIndex: number | null;
  hoveredIndex: number | null;
  disabled: boolean;
  onSelect: (index: number) => void;
  onHover: (index: number | null) => void;
  isLast: boolean;
  showTopLabel?: boolean;
  showBottomLabel?: boolean;
}

const OptionLabel = styled(Typography)<{ selected: boolean }>(({ theme, selected }) => ({
  color: theme.palette.grey[600],
  fontSize: theme.fixedSpacing(theme.tabiyaSpacing.sm * 1.2),
  fontWeight: selected ? "bold" : "normal",
  whiteSpace: "nowrap",
}));

const SkillsRankingVoteItem: React.FC<SkillsRankingVoteItemProps> = ({
  option,
  index,
  selectedIndex,
  hoveredIndex,
  disabled,
  onSelect,
  onHover,
  isLast,
  showTopLabel,
  showBottomLabel,
}) => {
  const theme = useTheme();
  const styleParams = { index, selectedIndex, hoveredIndex, disabled };
  const isSmallMobile = useMediaQuery((theme: Theme) => theme.breakpoints.down("sm"));

  return (
    <Box
      data-testid={DATA_TEST_ID.SKILLS_RANKING_VOTE_ITEM}
      display="flex"
      alignItems="center"
      width={isLast ? "auto" : "100%"}
    >
      <Box
        position="relative"
        display="flex"
        alignItems="center"
        flexDirection="column"
        sx={{
          opacity: disabled && selectedIndex !== index ? 0.5 : 1,
        }}
      >
        <Box
          sx={{ position: "absolute", top: -15, width: "max-content" }}
          data-testid={DATA_TEST_ID.SKILLS_RANKING_VOTE_ITEM_TOP_LABEL}
        >
          {showTopLabel && <OptionLabel selected={selectedIndex === index}>{option.value}</OptionLabel>}
        </Box>
        <PrimaryIconButton
          aria-label={option.value}
          onClick={() => onSelect(index)}
          onMouseEnter={() => onHover(index)}
          onMouseLeave={() => onHover(null)}
          disabled={disabled}
          sx={{ p: 0, minWidth: 0, minHeight: 0, borderRadius: "50%" }}
          data-testid={DATA_TEST_ID.SKILLS_RANKING_VOTE_ITEM_RADIO}
        >
          <span
            style={{
              display: "block",
              width: isSmallMobile ? 20 : 24,
              height: isSmallMobile ? 20 : 24,
              borderRadius: "50%",
              background: getBackgroundColor(styleParams, theme),
              border: getBorderStyle(styleParams, theme),
            }}
          />
        </PrimaryIconButton>
        <Box
          sx={{ position: "absolute", bottom: -15, width: "max-content" }}
          data-testid={DATA_TEST_ID.SKILLS_RANKING_VOTE_ITEM_BOTTOM_LABEL}
        >
          {showBottomLabel && <OptionLabel selected={selectedIndex === index}>{option.value}</OptionLabel>}
        </Box>
      </Box>
      {!isLast && (
        <Divider
          sx={{
            flex: 1,
            background: getDividerColor(styleParams, theme),
            borderWidth: 1,
            opacity: disabled ? 0.5 : 1,
          }}
          data-testid={DATA_TEST_ID.SKILLS_RANKING_VOTE_ITEM_DIVIDER}
        />
      )}
    </Box>
  );
};

export default SkillsRankingVoteItem;
