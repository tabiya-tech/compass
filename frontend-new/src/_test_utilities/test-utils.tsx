// Based on https://testing-library.com/docs/react-testing-library/setup/
import React, { ReactElement, ReactNode } from "react";
import { render, renderHook, RenderHookOptions, RenderOptions } from "@testing-library/react";
import { ThemeProvider } from "@mui/material";
import applicationTheme, { ThemeMode } from "src/theme/applicationTheme/applicationTheme";

// Import the Firebase mock utilities
import "src/_test_utilities/firebaseMock";
import SnackbarProvider from "src/theme/SnackbarProvider/SnackbarProvider";
import { IsOnlineProvider } from "src/app/isOnlineProvider/IsOnlineProvider";
jest.mock("firebase/compat/app", () => {
  return {
    initializeApp: jest.fn(),
    auth: jest.fn().mockReturnValue({
      signOut: jest.fn(),
      currentUser: {
        getIdToken: jest.fn(),
        getTokenResult: jest.fn(),
      },
      onAuthStateChanged: jest.fn().mockReturnValue(jest.fn()),
      signInWithPopup: jest.fn(),
      GoogleAuthProvider: { PROVIDER_ID: "google.com" },
    }),
  };
});

jest.mock("firebaseui", () => {
  return {
    auth: {
      AuthUI: {
        start: jest.fn(),
        getInstance: jest.fn().mockReturnValue({
          start: jest.fn(),
          reset: jest.fn(),
        }),
        reset: jest.fn(),
      },
    },
  };
});

jest.mock("@sentry/react", () => ({
  withSentry: (Component: any) => Component,
  getFeedback: jest.fn(),
  withProfiler: (Component: any) => Component,
  wrapCreateBrowserRouter: (Component: any) => Component,
}));

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <IsOnlineProvider>
      <ThemeProvider theme={applicationTheme(ThemeMode.LIGHT)}>
        <SnackbarProvider>{children}</SnackbarProvider>
      </ThemeProvider>
    </IsOnlineProvider>
  );
};

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, "wrapper">) =>
  render(ui, { wrapper: AllTheProviders, ...options });

function customRenderHook<T>(
  hook: () => T,
  options?: Omit<RenderHookOptions<T>, "wrapper"> & { wrapper?: React.ComponentType }
) {
  const CombinedWrapper = ({ children }: { children: ReactNode }) => {
    const Wrapper = options?.wrapper ?? React.Fragment;
    return (
      <AllTheProviders>
        <Wrapper>{children}</Wrapper>
      </AllTheProviders>
    );
  };

  return renderHook(hook, { wrapper: CombinedWrapper, ...options });
}

export * from "@testing-library/react";
export { customRender as render };
export { customRenderHook as renderHook };
