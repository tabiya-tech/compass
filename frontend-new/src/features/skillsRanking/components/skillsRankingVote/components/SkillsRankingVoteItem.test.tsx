// mute the console
import "src/_test_utilities/consoleMock";

import { render, screen } from "src/_test_utilities/test-utils";
import SkillsRankingVoteItem, {
  DATA_TEST_ID,
} from "src/features/skillsRanking/components/skillsRankingVote/components/SkillsRankingVoteItem";
import userEvent from "@testing-library/user-event";

describe("SkillsRankingVoteItem tests", () => {
  describe("render tests", () => {
    test("should render component successfully", () => {
      // GIVEN SkillsRankingVoteItem component
      const givenSkillsRankingVoteItem = (
        <SkillsRankingVoteItem
          option={{ value: "foo", percent: 10 }}
          index={0}
          selectedIndex={null}
          hoveredIndex={null}
          disabled={false}
          onSelect={jest.fn()}
          onHover={jest.fn()}
          isLast={false}
          showTopLabel={true}
          showBottomLabel={true}
        />
      );

      // WHEN the component is rendered
      render(givenSkillsRankingVoteItem);

      // THEN expect the container to be in the document
      const container = screen.getByTestId(DATA_TEST_ID.SKILLS_RANKING_VOTE_ITEM);
      expect(container).toBeInTheDocument();
      // AND expect the radio button to be in the document
      expect(screen.getByTestId(DATA_TEST_ID.SKILLS_RANKING_VOTE_ITEM_RADIO)).toBeInTheDocument();
      // AND expect the divider to be in the document
      expect(screen.getByTestId(DATA_TEST_ID.SKILLS_RANKING_VOTE_ITEM_DIVIDER)).toBeInTheDocument();
      // AND expect the top label to be in the document
      expect(screen.getByTestId(DATA_TEST_ID.SKILLS_RANKING_VOTE_ITEM_TOP_LABEL)).toBeInTheDocument();
      // AND expect the bottom label to be in the document
      expect(screen.getByTestId(DATA_TEST_ID.SKILLS_RANKING_VOTE_ITEM_BOTTOM_LABEL)).toBeInTheDocument();
      // AND to match the snapshot
      expect(container).toMatchSnapshot();
      // AND no errors or warnings to have occurred
      expect(console.error).not.toHaveBeenCalled();
      expect(console.warn).not.toHaveBeenCalled();
    });
  });

  describe("action tests", () => {
    test("should call onSelect with correct index when radio button is clicked", async () => {
      // GIVEN SkillsRankingVoteItem component
      const mockOnSelect = jest.fn();
      const givenIndex = 0;
      const givenSkillsRankingVoteItem = (
        <SkillsRankingVoteItem
          option={{ value: "foo", percent: 10 }}
          index={givenIndex}
          selectedIndex={null}
          hoveredIndex={null}
          disabled={false}
          onSelect={mockOnSelect}
          onHover={jest.fn()}
          isLast={false}
          showTopLabel={true}
          showBottomLabel={true}
        />
      );
      // AND the component is rendered
      render(givenSkillsRankingVoteItem);

      // WHEN the radio button is clicked
      const radioButton = screen.getByTestId(DATA_TEST_ID.SKILLS_RANKING_VOTE_ITEM_RADIO);
      await userEvent.click(radioButton);

      // THEN expect onSelect to have been called
      expect(mockOnSelect).toHaveBeenCalledTimes(1);
      // AND to have been called with the correct index
      expect(mockOnSelect).toHaveBeenCalledWith(givenIndex);
      // AND no errors or warnings to have occurred
      expect(console.error).not.toHaveBeenCalled();
      expect(console.warn).not.toHaveBeenCalled();
    });

    test("should call onHover with correct index when mouse enters and leaves the component", async () => {
      // GIVEN SkillsRankingVoteItem component
      const mockOnHover = jest.fn();
      const givenIndex = 0;
      const givenSkillsRankingVoteItem = (
        <SkillsRankingVoteItem
          option={{ value: "foo", percent: 10 }}
          index={givenIndex}
          selectedIndex={null}
          hoveredIndex={null}
          disabled={false}
          onSelect={jest.fn()}
          onHover={mockOnHover}
          isLast={false}
          showTopLabel={true}
          showBottomLabel={true}
        />
      );
      // AND the component is rendered
      render(givenSkillsRankingVoteItem);

      // WHEN the mouse enters the component
      const container = screen.getByTestId(DATA_TEST_ID.SKILLS_RANKING_VOTE_ITEM_RADIO);
      await userEvent.hover(container);

      // THEN expect onHover to have been called
      expect(mockOnHover).toHaveBeenCalledTimes(1);
      // AND to have been called with the correct index
      expect(mockOnHover).toHaveBeenCalledWith(givenIndex);

      // WHEN the mouse leaves the component
      await userEvent.unhover(container);

      // THEN expect onHover to have been called with null
      expect(mockOnHover).toHaveBeenCalledTimes(2);
      expect(mockOnHover).toHaveBeenLastCalledWith(null);
      // AND no errors or warnings to have occurred
      expect(console.error).not.toHaveBeenCalled();
      expect(console.warn).not.toHaveBeenCalled();
    });
  });
});
