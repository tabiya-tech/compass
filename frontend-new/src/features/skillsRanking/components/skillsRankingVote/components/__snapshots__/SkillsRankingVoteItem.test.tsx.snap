// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkillsRankingVoteItem tests render tests should render component successfully 1`] = `
<div
  class="MuiBox-root css-7pf6at"
  data-testid="skills-ranking-vote-item-5271610c-0000-4000-8000-000000000000"
>
  <div
    class="MuiBox-root css-1s0iju1"
  >
    <div
      class="MuiBox-root css-7csspk"
      data-testid="skills-ranking-vote-item-top-label-5271610c-0000-4000-8000-000000000000"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-18d7qa1-MuiTypography-root"
      >
        foo
      </p>
    </div>
    <button
      aria-label="foo"
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeMedium css-m3kom-MuiButtonBase-root-MuiIconButton-root"
      data-testid="skills-ranking-vote-item-radio-5271610c-0000-4000-8000-000000000000"
      tabindex="0"
      type="button"
    >
      <span
        style="display: block; width: 24px; height: 24px; border-radius: 50%; background: transparent; border: 2px solid rgb(0, 178, 101);"
      />
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <div
      class="MuiBox-root css-a3klcv"
      data-testid="skills-ranking-vote-item-bottom-label-5271610c-0000-4000-8000-000000000000"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-18d7qa1-MuiTypography-root"
      >
        foo
      </p>
    </div>
  </div>
  <hr
    class="MuiDivider-root MuiDivider-fullWidth css-19f4636-MuiDivider-root"
    data-testid="skills-ranking-vote-item-divider-5271610c-0000-4000-8000-000000000000"
  />
</div>
`;
